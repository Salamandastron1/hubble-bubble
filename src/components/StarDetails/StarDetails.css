.individual-card {
  display: grid;
  justify-items: center;
  grid-gap: 20px;
}
.individual-description {
  margin: 10px 50px 30px 50px;
  color: #e5c5c4;
  font-family: 'Roboto';
  font-size: 1.5em;
}
.individual-name {
  font-family: 'Titan One';
  color: #e5c5c4;
  font-size: 3.5em;
  margin: 0;
}
.back-btn {
  color: #e5c5c4;
}