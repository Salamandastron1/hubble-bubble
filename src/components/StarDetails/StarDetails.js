import React from 'react'
import './StarDetails.css'
import { Link } from 'react-router-dom'

const StarDetails = (props) => {
  const {image_files, description, name} = props
  return(
    <section>
      <Link to='/studytime' className='back-btn'>â—€ back</Link>
      <h2>{name}</h2>
      <img src={image_files} alt ={name}/>
      <p>{description} </p>
    </section>
  )
}

export default StarDetails;