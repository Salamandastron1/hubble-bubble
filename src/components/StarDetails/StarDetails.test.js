import React from 'react'
import { shallow } from 'enzyme'
import {StarDetails, mapStateToProps} from './StarDetails'

describe('StarDetails', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow (<StarDetails />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with the store state', () => {
    const desired = {
      isLoading: true
    }
    const mockState = {
      isLoading: true,
      error: 'meow',
      filter: 'show'
    }
    const result = mapStateToProps(mockState)

    expect(result).toEqual(desired)
  })
})