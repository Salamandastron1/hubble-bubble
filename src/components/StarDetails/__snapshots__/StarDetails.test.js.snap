// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarDetails should match the snapshot 1`] = `
<section
  className="individual-card"
>
  <Link
    className="back-btn"
    replace={false}
    to="/hubble-bubble/studytime"
  >
    â—€ back
  </Link>
  <h2
    className="individual-name"
  />
  <img
    height="500"
    width="500"
  />
  <p
    className="individual-description"
  />
</section>
`;
